src/libs/common/types/Selection.ts

```diff
@@ -0,0 +1,62 @@
+import { Position, Range } from "@cursorless/common";
+
+export class Selection extends Range {
```

Comment:
How do we want to handle equality for selections?  On the one hand, it doesn't seem right for two selections to be equal if their anchor and active are swapped.  On the other hand, a client might only know that they're instances of `Range`, and be surprised when they have the same start and end, but are not equal 🤔

Kinda argues for selection not to derive from range, and instead just support a `selection.range` accessor

Reply:
We should jot compare anchor and active

Reply:
why?

Reply:
Talon that can't hear the word just apparently

Reply:
Ok, I saw your change.  You don't think the following would be surprising:

```typescript
function foo(rangeA: Range, rangeB: Range) {
    if (rangeA.isEqual(rangeB)) {
        ...
    }
}

...

foo(selectionA, selectionB)
```

And then even though they are the same range, that check returns false

Reply:
I'm going to quickly remove that `extends Range` and just see how much breaks.  If it's too much stuff, then let's call it out of scope, because we're mirroring VSCode behaviour for now

Reply:
Ok well that was a disaster.  What if we just rename `isEqual` to `isRangeEqual`?  It is used in exactly 2 places in our entire codebase 🤔

Reply:
Of course it can be but so far we haven't really run into that. In Cursorless we are barely using selections at all since we prefer a range and a boolean instead. 

Sounds reasonable

Reply:
Yeah I think it is useful, but the semantics are pretty odd in this case.  I'd be inclined to use `isRangeEqual` for now and then when we need it we can add a better `isEqual`, with a JSDoc comment indicating why this function might not do what you expect.  Make sense?

Reply:
Go ahead